lib/pbench/test/unit/server/test_indexing_tarballs.py

```diff
@@ -290,22 +413,93 @@ def test_process_tb_none(self, mocks, index):
             and not FakePbenchTarBall.make_tool_called
         )
 
-    def test_process_tb(self, mocks, index):
+    def test_process_tb_bad_load(self, mocks, index):
+        FakePbenchTemplates.failure = Exception("I think I can't")
+        stat = index.process_tb(tarballs=[])
+        assert stat == 12
+        assert FakePbenchTemplates.templates_updated
+
+    def test_process_tb_term(self, mocks, index):
+        def fake_es_index(es, actions, errorsfp, logger, _dbg=0):
+            raise SigTermException("ter-min-ate; ter-min-ate")
+
+        mocks.setattr("pbench.server.indexing_tarballs.es_index", fake_es_index)
+        stat = index.process_tb(tarballs=[tarball_2, tarball_1])
+        assert stat == 0
+
+    def test_process_tb_interrupt(self, mocks, index):
+        def fake_es_index(es, actions, errorsfp, logger, _dbg=0):
+            raise SigTermException("ter-min-ate; ter-min-ate")
+
+        mocks.setattr("pbench.server.indexing_tarballs.es_index", fake_es_index)
+        stat = index.process_tb(tarballs=[tarball_2, tarball_1])
+        assert stat == 0
+
+    def test_process_tb_int(self, mocks, index):
+        """Test behavior when a SIGHUP occurs during processing.
+
+        This should trigger the indexer to re-evaluate the set of enabled
+        datasets after completing the current dataset. Because the mock
+        here generates a SIGHUP on the first index operation any additional
+        enabled datasets would be missed unless they're still enabled and
+        are reported by the subsequent collect_tb() call. The verified action
+        sequence here confirms that the second (skipped) dataset in the
+        initial parameter list (tarball_1) is only indexed once despite also
+        appearing in the SIGHUP collect_tb.
+        """
         index_actions = []
+        first_index = True
 
         def fake_es_index(es, actions, errorsfp, logger, _dbg=0):
+            nonlocal first_index
+            if first_index:
+                first_index = False
+                os.kill(os.getpid(), SIGHUP)
             index_actions.append(actions)
             return (1000, 2000, 1, 0, 0, 0)
 
         mocks.setattr("pbench.server.indexing_tarballs.es_index", fake_es_index)
-        ds1 = FakeDataset(name="ds1", resource_id="ABC")
-        ds2 = FakeDataset(name="ds2", resource_id="ACDF")
-        stat = index.process_tb(
-            tarballs=[
-                TarballData(dataset=ds2, size=2, tarball=f"{ds2.name}.tar.xz"),
-                TarballData(dataset=ds1, size=5, tarball=f"{ds1.name}.tar.xz"),
-            ]
+        mocks.setattr(Index, "collect_tb", lambda self: (0, [tarball_1, tarball_3]))
+        stat = index.process_tb(tarballs=[tarball_2, tarball_1])
+        assert (
+            stat == 0
+            and FakePbenchTarBall.make_all_called == 3
+            and not FakePbenchTarBall.make_tool_called
         )
```

Comment:
It seemed there was no particular reason to separate them, either. Yeah, it would avoid the parens. If I make another pass, and remember, maybe I'll change it, but I don't feel much urgency.